Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -499,8 +535,8 @@ void AODProducerWorkflowDPL::addToFwdTracksTable(FwdTracksCursorType& fwdTracksC
     chi2matchmchmid = track.getMIDMatchingChi2();
     chi2matchmchmft = track.getMFTMCHMatchingChi2();
     matchscoremchmft = track.getMFTMCHMatchingScore();
-    matchmfttrackid = track.getMFTTrackID();
-    matchmchtrackid = track.getMCHTrackID();
+    matchmfttrackid = mIndexTableMFT[track.getMFTTrackID()];
+    matchmchtrackid = mIndexTableFwd[track.getMCHTrackID()];
```

Comment:
also here: mIndexTableFwd is filled in the fillIndexTablesPerCollision only for standalone MCH tracks, while here you are using it as remapping for matched one.

Reply:
Here I use the map mIndexTableFwd to retrieve the correct indices for MCH standalone, since 'matchmchtrackid' is supposed to be pointing to a MCH track in the table.

Reply:
I mean the only place you are filling the `mIndexTableFwd` is https://github.com/AliceO2Group/AliceO2/blob/b2699abec433f6ed70585650719058455cd61e1f/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L377-L379, i.e. only for standalone MCH tracks but not for the MCH contributor of the matched track. Because of the way the CreateTracksVariadic treats forward tracks, this may work now (since the MCH tracks which are matched are present also in the standalone MCH tracks pool, so you effectively account them in the mIndexTableFwd) but if you switch to non-redundant version, the mIndexTableFwd for matched MCH track will not be filled.